framework/source/class/qx/event/message/Bus.js

```diff
@@ -191,30 +198,52 @@ qx.Class.define("qx.event.message.Bus",
       }
     },
 
+    /**
+     * Subscribes to a topic just for one dispach and automatically unsubscribe
+     * afterwards.
+     *
+     * @param topic {String|RegExp} Topic to subscribe to. see {@link qx.event.message.Bus#subscribe}
+     * for details
+     * @param subscriber {Function} Message handler function
+     * @param context {Object} The execution context of the callback (i.e. "this")
+     * @return {Boolean} Success
+     */
+    subscribeOnce : function(topic, subscriber, context)
+    {
+      var that = this;
+      var modified_subscriber = function(msg) {
+        subscriber.call(context, msg);
+        that.unsubscribe(topic, modified_subscriber, context);
+      }
+      this.subscribe(topic, modified_subscriber, context);
```

Comment:
This makes it impossible to unsubscribe because the user has no access to `modified_subscriber`. That may be enough of a downside to negate the usefulness of this new feature. Probably `subscribeOnce` should be left as an exercise for the user who needs to unsubscribe after a single use, by simply adding the `unsubscribe` call to their handler.

Reply:
What would be a situation in which a self-unsubscribing subscriber would have to unsubscribed? The very point of this method is that you don't have to unsubscribe it - so the user doesn't need to have access to the outer function. I can certainly add a note to the documentation that it is not possible to unsubscribe, but as I said I wouldn't know why anyone would need to do that?

On the other hand, I added the method because writing self-unsubscribing message handlers is annoying ...

Reply:
An example would be that you set up something on the assumption that your app will initialize properly, but something times out and you now don't want that subscription to fire. I guess that sounds made up, but all of the times that you might unsubscribe from any other type of subscription would apply here as well.

Your solution of just documenting the fact that you can't unsubscribe seems very adequate.

Reply:
Thanks for alerting me to this.